pytorch_lightning/loops/fit_loop.py

```diff
@@ -278,7 +278,7 @@ def should_accumulate(self) -> bool:
     def _check_checkpoint_callback(self, should_update: bool, is_last: bool = False):
         """Checks if checkpointing needs to be done"""
         # TODO: bake this logic into the ModelCheckpoint callback
-        if should_update and self.trainer.checkpoint_connector.has_trained:
```

Comment:
I think this was here to prevent checking the callback if no training steps were made (i.e., training immediately got interrupted)

Reply:
I get that but CI passes after removing it.

Tests were added in #4291 and #4372.

Reply:
ok, looks good! 

I usually don't trust tests, especially when removing something from the machine and it still appears to be working ^^

Reply:
I think I added `self.trainer.checkpoint_connector.has_trained` to make sure we don't increment `current_epoch` if training hasn't been called while saving a checkpoint.
I believe we should have a test for it, but I am surprised it doesn't fail :) 
I would align with Adrian there and I won't believe our tests ahah

Reply:
It's hard for me to write a test when I don't know what I should be testing exactly.

But some analysis:

This flag is only used here in `_check_checkpoint_callback`.
`_check_checkpoint_callback` is only called from `FitLoop.on_advance_end` and `FitLoop.on_run_end`.
`on_advance_end` is only called if it has trained by design (see `Loop.run`)
`on_run_end` can be called if the first iteration of `FitLoop.done` returns `True`. We can replicate it like this:

```python
def test_has_trained(tmpdir):
    class TestModel(BoringModel):
        def on_train_start(self) -> None:
            self.trainer.should_stop = True

    model = TestModel()
    mc = ModelCheckpoint(dirpath=tmpdir)
    trainer = Trainer(default_root_dir=tmpdir, min_epochs=0, logger=False, callbacks=[mc])
    trainer.fit(model)

    assert os.listdir(tmpdir) == []
```

However, the test passes because even though it tries to save here when it hasn't trained, the global step check here avoids it:

https://github.com/PyTorchLightning/pytorch-lightning/blob/f6a5bb2eee18e23c1a15013750c82539f399e9da/pytorch_lightning/callbacks/model_checkpoint.py#L344

Let me know if some other possibility should be explored or what would be a better test

Reply:
In the trainer we had this manual checkpoint saving call outside the run, in the exception handling. I guess it's because of that right? I'm not sure